### ISSUE TITLE
Shorter mangled names for common punctuation

### ISSUE THREAD

**Issue by gilch** [hylang/hy#1577]
Some common Hy symbols become very long-winded when mangled.
```
=> (mangle '->>)
'_hyx_XgreaterHthan_signXXgreaterHthan_signX'
```
This was much easier to read when we just put `->>` in the ast. It can't be that nice if we want a valid Python identifier, but I think we could clean this up. I'm not suggesting we go back to meaningless codes, but we could have a secondary lookup table of shorter names for the punctuation we'd commonly see (i.e. ASCII). On the US keyboard, we have ``` `~!@#$%^&*()_+-={}|[]\:";'<>?,./```

| ? |Unicode name | short name|
|-|---------------------|----------|
|\`| Xgrave_accentX      | XtickX   |
|!| Xexclamation_markX  | XbangX   |
|@| Xcommercial_atX     | XatX     |
|#| Xnumber_signX       | XtagX   |
|$| Xdollar_signX       | XsbarX |
|%| Xpercent_signX      | XmodX  |
|^| Xcircumflex_accentX | XhatX    |
|&| XampersandX         | XetX    |
|*| XasteriskX          | XstarX   |
|+| Xplus_signX         | XaddX   |
|=| Xequals_signX       | XeqX |
|\ | Xreverse_solidusX   | XbslashX |
|\|| Xvertical_lineX     | XorX    |
|< | XlessHthan_signX    | XltX     |
|> | XgreaterHthan_signX | XgtX     |
|? | Xquestion_markX     | XqueryX  |
|. | Xfull_stopX         | XdotX |
|/ | XsolidusX           | XperX  |
|X| Xlatin_capital_letter_xX | XX |
|- | XhyphenHminusX      | XsubX  |
|_ |Xlow_lineX | XscoreX |

The other ASCII punctuation marks aren't lexically valid in symbols, or already had a short enough Unicode name (XcommaX, XcolonX. And XsolidusX was _almost_ short enough). Thus, e.g., `->>` would mangle to the much shorter `_hyx_XgtXXgtX`.

We can bikeshed the short names a bit, but I think the above is reasonable. They're either readable common names or the same names Python uses in the `operator` module and in dunder names, which makes it less arbitrary.

[Edit: changed XminusX to XsubX and XplusX to XaddX and XequalsX to XeqX to fit better with Python operator naming conventions. Changed XperiodX to XdotX because it's shorter.

I'm less certain about the rest.
Changed XandX to XetX, which is what the & ligature means anyway. Changed XdollarX to XsbarX because it's shorter. Changed XslashX to XperX. Changed XhashX to XtagX.]

**Comment by Kodiologist** [comment]
Clojure also has some short names it uses to munge punctuation characters, if I recall. We could use those.

**Comment by gilch** [comment]
It does
```
user=> (munge (apply str (vec (map char (range 33 126)))))
"_BANG__DOUBLEQUOTE__SHARP_$_PERCENT__AMPERSAND__SINGLEQUOTE_()_STAR__PLUS_,_._SLASH_0123456789_COLON_;_LT__EQ__GT__QMARK__CIRCA_ABCDEFGHIJKLMNOPQRSTUVWXYZ_LBRACK__BSLASH__RBRACK__CARET__`abcdefghijklmnopqrstuvwxyz_LBRACE__BAR__RBRACE_"
```
Re-using Clojure's choices might be less arbitrary, but it doesn't fit all that well with Python and Hy. It's more ideas, anyway. I don't think we need a short name for brackets and quotes, since those aren't allowed in lexically-valid symbols anyway. Note that `!*+/:<>\` already have the same names in my suggested table. Notable differences are _SHARP_/XhashX, _PERCENT_/XmodX, _AMPERSAND_/XandX, _EQ_/XequalsX, _QMARK_/XqueryX, _CIRCA_/XatX, _CARET_/XhatX, and _BAR_/XorX. I do like mine better here, except maybe XeqX and XbarX also seem OK. Python does have an `eq` operator. I almost went with XqmarkX, XeqX, and XbarX and I also considered XpipeX for `|`. I think XpercentX and XampersandX would be too long. `&` is an operator, and it's used in `&rest` and friends. `%` is also an operator and it's used in the `#%` macro for arguments. The `%**` argument would be particularly long. `hyx_Xpercent_signXXasteriskXXasteriskX` vs `hyx_XmodXXstarXXstarX`.

**Comment by gilch** [comment]
Note also that Clojure doesn't munge `,` (whitespace in Clojure, thus not in symbols) and `$` (valid in Java identifiers, but not Python) but Hy would have to mangle those. We can't just lift it wholesale from Clojure.

**Comment by gilch** [comment]
I think changing XminusX to XsubX and XplusX to XaddX fits better with Python conventions. Clojure used `_PLUS_` though. And `-_` names are moot if they're both mangled as `_`, but we'd have to distinguish them if we want full reversibility, which is a separate issue.

**Comment by gilch** [comment]
Short names are a lot more desirable, but not if we can't read them. "period", "bslash" and "dollar" are six characters each. I shortened "period" to "dot".

Dollar could be shortened to "sbar", "buck", "bill", "coin", or "dolr".

"query", "score", "slash" are five each.

"slash" could be "over" or "per". I don't know what to do about "bslash". Maybe "esc", but that might be confused with the control character. "under" is not much shorter. Maybe "bs". It's just not used that much though.

"query" could be "eh" or "hook". We've already got the `is_` prefix, which takes care of the common case with a short answer, but I'm still kind of uncomfortable with the special treatment.

Even some of the shorter ones could be shorter. "and" could be "et". "hash" could be "sh".
